sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -13,24 +13,104 @@ import { WebResource } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
 
+function environmentProxyValue(name: string): string | undefined {
+  if (process.env[name]) {
+    return process.env[name];
+  } else if (process.env[name.toLowerCase()]) {
+    return process.env[name.toLowerCase()];
+  }
+  return undefined;
+}
+
 function loadEnvironmentProxyValue(): string | undefined {
+  let httpProxy = null;
+  let httpsProxy = null;
+  let allProxy = null;
+
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  httpsProxy = environmentProxyValue(Constants.HTTPS_PROXY)!;
+  httpProxy = environmentProxyValue(Constants.HTTP_PROXY)!;
+
+  if (httpProxy == null || httpsProxy == null) {
+    allProxy = environmentProxyValue(Constants.ALL_PROXY)!;
+
+    if (httpProxy == null) {
+      httpProxy = allProxy;
+    } else {
+      httpsProxy = allProxy;
+    }
+  }
+
+  if (httpProxy == null && httpsProxy == null) {
+    return undefined;
+  }
+
+  if (httpsProxy) {
+    return httpsProxy;
+  } else if (httpProxy) {
+    return httpProxy;
   }
 
   return undefined;
 }
 
+export function isBypassed(uri: string) {
+  let bypass = loadNoProxy();
+  let flag = false;
+  if (bypass != null) {
+    bypass.forEach((s) => {
+      if (s[0] == ".") {
+        if (uri.endsWith(s)) {
+          flag = true;
+        } else {
+          uri = getHost(uri);
+          if (uri == s.slice(1) && uri.length == s.length - 1) {
+            flag = true;
+          }
+        }
+      } else {
+        if (uri == s) {
+          flag = true;
+        }
+      }
+    });
+    return flag;
+  }
+  return false;
+}
+
+function getHost(uri: string) {
+  if (uri.toString().startsWith("http://")) {
+    uri = uri.toString().slice(7);
+  }
+  if (uri.toString().startsWith("https://")) {
+    uri = uri.toString().slice(8);
+  }
+  return uri;
+}
+
+function loadNoProxy() {
+  let noProxy = environmentProxyValue(Constants.NO_PROXY)!;
+  if (noProxy != null) {
+    let list = noProxy.split(",");
+    let tmpList = new Array();
+    list.forEach((value) => {
+      let tmp = value.trim();
+      if (tmp.length > 0) {
+        tmpList.push(tmp);
+      }
+    });
+
+    if (tmpList.length > 0) {
+      return tmpList;
+    }
+  }
+  return null;
```

Comment:
why not just return the empty array? the for loop over it will just never iterate and not hurt anything